geopandas/geodataframe.py

```diff
@@ -1441,6 +1435,27 @@ def apply(self, func, axis=0, raw=False, result_type=None, args=(), **kwargs):
 
     @property
     def _constructor(self):
+        geometry_default = self._geometry_column_name
+        crs_default = self.crs
+
+        def _geodataframe_constructor_with_fallback(
```

Comment:
In the geoseries PR you suggested not defining the analogue to this function as a method closure. I could also do that here if that makes sense `_geodataframe_constructor_with_fallback` would then take crs_default and geometry_default as additional arguments, and `crs` and `geometry` would default to `NO_VALUE_GIVEN = object()`, and then be replaced with the default values. I think we also had a similar discussion elsewhere that I now can't find, if this is only going to be called by pandas internally, then crs and geometry will never be supplied, so I could drop them as args to simplify further.